File: phase00-foundations/two_sum.js
// Problem: Two Sum
// Approach: Brute Force

const twoSum = (arr_nums, target) => {
  for (let i = 0; i < arr_nums.length; i++) {
    for (let j = i + 1; i < arr_nums.length; j++) {
      if (arr_nums[i] + arr_nums[j] === target) {
        return [i, j];
      }
    }
  }
};
const nums = [2, 7, 11, 15];
const target = 9;

result = twoSum(nums, target);
console.log(result);

File: base-track/phase01-arrays-and-hashing/two_sum.py
# Problem: Two Sum
# Difficulty: Easy
# Link: https://leetcode.com/problems/two-sum/
# Date: 15-06-2026
# Approach: hashmap - store complement as we iterate
# Time: O(n) | Space: O(n)


def twoSum(nums, target):
    seen = {}

    for i in range(0, len(nums)):
        n = target - nums[i]
        if n in seen:
            return [i, seen[n]]
        seen[nums[i]] = i


nums = [2, 7, 11, 15]
print(twoSum(nums, 26))
